lib/mayaUsd/render/vp2ShaderFragments/usdPreviewSurfaceLightingAPI3.xml

```diff
@@ -0,0 +1,1132 @@
+             clearcoatColor,              // Clearcoat color 90
+             clearcoatRoughness,          // Roughness
+             fresnel,                     // Fresnel
+             NdotL, NdotE, NdotH, EdotH); // Dot products needed for lights
+    }
+
+    LightingContributions lightingContrib;
+
+    lightingContrib.diffuse =
+        occlusion * NdotL * d * lightDiffuseIrradiance;
+
+    lightingContrib.specular =
+        occlusion * NdotL * (s1 + s2) * lightSpecularIrradiance;
+
+    return lightingContrib;
+}
+
+
+//-- glsl Preview.LightIntegration
+
+vec3
+evaluateIndirectLighting(
+        vec3 diffuseColor,
+        vec3 specularColor,
+        vec3 Neye,
+        vec3 E,
+        float EdotH,
+        float ior,
+        float metallic,
+        float occlusion,
+        float roughness,
+        bool useSpecularWorkflow,
+        float clearcoatAmount, 
+        vec3 clearcoatColor, 
+        float clearcoatRoughness)
+{
+    vec3 indirect = vec3(0.0);
+
+    vec3 F0 = specularColor;
+    vec3 F90 = vec3(1.0);
+    vec3 d = diffuseColor;
+    float R = (1.0 - ior) / (1.0 + ior);
+    if (!useSpecularWorkflow) {
+        vec3 specColor = mix(vec3(1.0), diffuseColor, metallic);
+        F0 = mix(R * R * specColor, specColor, metallic);
+        F90 = specColor;
+
+        // For metallic workflows, pure metals have no diffuse
+        d *= 1.0 - metallic;
+    }
+    // Adjust the diffuse so glazing angles have less diffuse
+    float fresnel = SchlickFresnel(EdotH);
+    vec3 F = mix(F0, F90, fresnel);
+    d *= (1.0 - F);
+
+    vec3 diffuse = mayaGetIrradianceEnvironment(Neye);
+
+    // Specular Component 
+    float phongExp = mayaRoughnessToPhongExp(roughness);
+    vec3 Li = mayaGetSpecularEnvironment(Neye, E, phongExp);
+    vec3 specular = Li * F;
+
+    // Clearcoat Component
+    vec3 clearcoat = vec3(0.0);
+    if (clearcoatAmount > 0.0) {
+        vec3 clearcoatF = clearcoatAmount * mix(
+            R * R * clearcoatColor, // Clearcoat F0
+            clearcoatColor,         // Clearcoat F90
+            fresnel);
+        phongExp = mayaRoughnessToPhongExp(clearcoatRoughness);
+        Li = mayaGetSpecularEnvironment(Neye, E, phongExp);
+        clearcoat = Li * clearcoatF;
+    }
+
+    // Indirect Lighting
+    indirect = (d * diffuse + specular + clearcoat) * occlusion;
+
+    return indirect;
+}
+
+vec3
+evaluateLights(
+        vec3 emissiveColor,
+        vec3 diffuseColor,
+        bool useSpecularWorkflow,
+        float ior,
+        float metallic,
+        float specularAmount,
+        vec3 specularColor,
+        float specularRoughness,
+        float clearcoatAmount,
+        vec3 clearcoatColor,
+        float clearcoatRoughness,
+        float occlusion,
+        vec3 Peye,
+        vec3 Neye)
+{
+    vec3 n = Neye;
+    vec3 e = normalize(u_viewPosition - Peye);
+    float NdotE = max(0.0, dot(n, e));
+
+    vec3 directLight = diffuseColor * mayaGetAmbientLight();
```

Comment:
One line was changed here compared to API2 which initialized to vec3(0,0,0) instead of looking up for a missing ambient light.